### ISSUE TITLE
OptunaSearchCV: No option if scoring needs to be maximized or minimized

### ISSUE THREAD

**Issue by kienerj** [optuna/optuna#2948]
## Motivation

`OptunaSearchCV` allows to set a scoring function/string. However there is no option to tell it if the score needs to be minimized or maximized.

## Description

Add an additional argument to OptunaSearchCV that tells optuna if the score should be maximized or minimized.
(like in rays `TuneSearchCV` where such a `mode` argument exists.)

**Comment by nzw0301** [comment]
Thank you for creating a feature request! In my understanding, all scikit-learn's scoring functions 

> follow the convention that higher return values are better than lower return values

according to [this page](https://scikit-learn.org/stable/modules/model_evaluation.html).

So when we define a callable function, we might need to want to specify `minimize` for study's direction. But in this case, we can use `maximize` direction by multiplying the scoring value by `-1` in the callable function in practice. What do you think about this?

**Comment by nzw0301** [comment]
I suppose that it would be great to mention this convention on `OptunaSearchCV` page because the default direction of Optuna is `minimize` but `OptunaSearchCV`'s direction is`maximize`.

**Comment by kienerj** [comment]
Yes I agree with mentioning it on the documentation. That should make it clearer what is expected.
